<!DOCTYPE html>
<html lang="en">
<head>

  <!-- Google tag (gtag.js) -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=G-5RD3Q65E7R"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'G-5RD3Q65E7R');
  </script>

  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">

  <!-- Favicon code from realfavicongenerator.net -->
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#8b51a3">
<meta name="msapplication-TileColor" content="#563d7c">
<meta name="theme-color" content="#ffffff">

  <!--jQuery-->
<script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>

  <!-- Fonts & Icons -->
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous">
  <link href='//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css' rel='stylesheet' type='text/css'>

  <!-- CSS -->
  <link rel="stylesheet" href="/assets/css/main.css">

  <!-- Begin Jekyll SEO tag v2.8.0 -->
<title>Publications | Cris Taniguti</title>
<meta name="generator" content="Jekyll v4.3.3" />
<meta property="og:title" content="Publications" />
<meta name="author" content="Cris Taniguti" />
<meta property="og:locale" content="en_GB" />
<meta name="description" content="List of publications" />
<meta property="og:description" content="List of publications" />
<link rel="canonical" href="http://localhost:4000/publications.html" />
<meta property="og:url" content="http://localhost:4000/publications.html" />
<meta property="og:site_name" content="Cris Taniguti" />
<meta property="og:type" content="website" />
<meta name="twitter:card" content="summary" />
<meta property="twitter:title" content="Publications" />
<script type="application/ld+json">
{"@context":"https://schema.org","@type":"WebPage","author":{"@type":"Person","name":"Cris Taniguti"},"description":"List of publications","headline":"Publications","publisher":{"@type":"Organization","logo":{"@type":"ImageObject","url":"http://localhost:4000/assets/img/smile.png"},"name":"Cris Taniguti"},"url":"http://localhost:4000/publications.html"}</script>
<!-- End Jekyll SEO tag -->


</head>

<!--jQuery-->
<script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script>
<body>
  <div class="container">
    

<section id="header-nav">
  <header>
    <nav>
      <ul>
        
        <a href="/">
          <li class="btn-nav">Home</li>
        </a>
        <li class="current">Publications</li>
        
          <a href="/presentations">
            <li class="btn-nav">Presentations</li>
          </a>
        
        
          <a href="/blog">
            <li class="btn-nav">Blog</li>
          </a>
          <a href="/tags">
            <li class="btn-nav">Tags</li>
          </a>
        
        <!-- presentations -->
        
      </ul>
    </nav>
  </header>
</section>
<!--publications.js-->
<script type="text/javascript" src="/assets/js/publications.js"></script>
<div id="publications">
  <section class="bg"></section>
  <h1 class="title">Publications</h1>
  <h2 class="bibliography">2025</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Taniguti2025">
  
  <!-- Title -->
  <div class="title">Exploring chromosomal variations in garden roses: Insights from high‐density SNP array data and a new tool, Qploidy</div>
  <!-- Author -->
  <div class="author">Taniguti, Cristiane H.,&nbsp;Lau, Jeekin,&nbsp;Hochhaus, Tessa,&nbsp;Arias, Diana C. Lopez,&nbsp;Hokanson, Stan C.,&nbsp;Zlesak, David C.,&nbsp;Byrne, David H.,&nbsp;Klein, Patricia E.,&nbsp;and Riera‐Lizarazu, Oscar
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    The Plant Genome, vol. 18, Jun, 2025
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="https://acsess.onlinelibrary.wiley.com/doi/10.1002/tpg2.70044" class="btn btn-sm z-depth-0" role="button">URL</a>]
      [<a href="http://doi.org/10.1002/tpg2.70044" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p><p> Roses ( <italic>Rosa</italic> L.) are among the most economically important ornamentals worldwide, with ploidy ranging from diploid (2x) to hendecaploid (11x), though most cultivars are diploid (2x), triploid (3x), or tetraploid (4x). To enable large‐scale analyses of ploidy and aneuploidy in roses using high‐density single nucleotide polymorphism (SNP) array data, we developed Qploidy, an R package. Qploidy leverages tools for estimating allele dosage, adapts methods from human genetics for copy number estimation, and optimizes the standardization of allele intensity ( <italic>R</italic> ) and B allele frequency (BAF) for ploidy levels greater than 2x. With Qploidy, we analyzed a dataset of 1944 samples consisting of 588 samples from a germplasm collection and 1356 samples from 14 biparental mapping populations. The majority of genotypes in the germplasm collection were tetraploid (56%), followed by diploids (20%) and triploids (11%). The percentage of aneuploids was lower in the germplasm collection (2%) compared to biparental populations (16%). Reduced fitness likely explains the higher frequency of aneuploids in mapping populations compared to the germplasm collection, where stronger selective pressures normally act. In tetraploid biparental populations, pentasomy (65%) was significantly more common than trisomy (19%). Also, aneuploid states were predominantly transmitted through the female parent (87%), suggesting greater gametophyte sensitivity to chromosome number variation, particularly the loss of a chromosome. Since aneuploidy disturbs linkage and quantitative trait loci (QTL) analyses, Qploidy may also be used to guide the removal of aneuploid‐affected data prior to downstream analysis. Besides roses, Qploidy can be used to study ploidy and aneuploidy in other polyploid species. </p></p>
    </div>
</div></li>
<li><!-- Entry bib key -->
<div id="Sandercock2025">
  
  <!-- Title -->
  <div class="title">BIGapp: A user‐friendly genomic tool kit identified quantitative trait loci for creeping rootedness in alfalfa ( <i>Medicago sativa</i> L.)</div>
  <!-- Author -->
  <div class="author">Sandercock, Alexander M.,&nbsp;Peel, Michael D.,&nbsp;Taniguti, Cristiane H.,&nbsp;Chinchilla‐Vargas, Josué,&nbsp;Chen, Shufen,&nbsp;Sapkota, Manoj,&nbsp;Lin, Meng,&nbsp;Zhao, Dongyan,&nbsp;Ackerman, Arlyn J.,&nbsp;Basnet, Bhoja R.,&nbsp;Beil, Craig T.,&nbsp;and Sheehan, Moira J.
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    The Plant Genome, vol. 18, Sep, 2025
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="https://acsess.onlinelibrary.wiley.com/doi/10.1002/tpg2.70067" class="btn btn-sm z-depth-0" role="button">URL</a>]
      [<a href="http://doi.org/10.1002/tpg2.70067" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p><p> Alfalfa ( <italic>Medicago sativa</italic> L.) is a globally vital forage crop valued for its perennial growth and multiple annual harvests. A breeding effort is underway to improve the crop for productivity and persistence against biotic and abiotic stresses using “creeping rootedness,” a trait where plants exhibit horizontal root growth, similar to rhizomes, with increased vegetative ground surface area. In this study, we genotyped a breeding population of 648 alfalfa lines segregating for creeping rootedness using the 3K DArTag marker panel to identify trait‐associated genomic loci and evaluate the feasibility of genomic prediction to accelerate breeding cycles. Using genome‐wide association studies (GWAS), we identified three quantitative trait loci (QTLs), with one major QTL located on chromosome 6.1 associated with this trait. Genomic prediction showed moderate predictive ability ( <italic>r</italic>  = 0.68) for creeping rootedness. A significant advancement in this study was the development and utilization of the Breeding Insight Genomics Application (BIGapp), an R Shiny application designed to streamline the processing of genomic data through an intuitive interface. This tool makes integrating genomics into existing breeding programs accessible, regardless of species ploidy or the researcher’s coding proficiency. The identified QTL will be essential in future efforts to develop new alfalfa cultivars with the creeping rootedness trait and accelerate the breeding cycle, with BIGapp playing a pivotal role in these advancements. </p></p>
    </div>
</div></li></ol>
<h2 class="bibliography">2023</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Lau2023">
  
  <!-- Title -->
  <div class="title">QTL discovery for resistance to black spot and cercospora leaf spot, and defoliation in two interconnected F1 bi-parental tetraploid garden rose populations</div>
  <!-- Author -->
  <div class="author">Lau, Jeekin,&nbsp;Gill, Haramrit,&nbsp;Taniguti, Cristiane H.,&nbsp;Young, Ellen L.,&nbsp;Klein, Patricia E.,&nbsp;Byrne, David H.,&nbsp;and Riera-Lizarazu, Oscar
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Frontiers in Plant Science, vol. 14, 2023
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.3389/fpls.2023.1209445" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Garden roses are an economically important horticultural crop worldwide, and two major fungal pathogens, black spot (Diplocarpon rosae F.A. Wolf) and cercospora leaf spot of rose (Rosisphaerella rosicola Pass.), affect both the health and ornamental value of the plant. Most studies on black spot disease resistance have focused on diploid germplasm, and little work has been performed on cercospora leaf spot resistance. With the use of newly developed software tools for autopolyploid genetics, two interconnected tetraploid garden rose F1 populations (phenotyped over the course of 3 years) were used for quantitative trait locus (QTL) analysis of black spot and cercospora leaf spot resistance as well as plant defoliation. QTLs for black spot resistance were mapped to linkage groups (LGs) 1–6. QTLs for cercospora resistance and susceptibility were found in LGs 1, 4, and 5 and for defoliation in LGs 1, 3, and 5. The major locus on LG 5 for black spot resistance coincides with the previously discovered Rdr4 locus inherited from Rosa L. ‘Radbrite’ (Brite Eyes™), the common parent used in these mapping populations. This work is the first report of any QTL for cercospora resistance/susceptibility in tetraploid rose germplasm and the first report of defoliation QTL in roses. A major QTL for cercospora susceptibility coincides with the black spot resistance QTL on LG 5 (Rdr4). A major cercospora resistance QTL was found on LG 1. These populations provide a genetic resource that will further the knowledge base of rose genetics as more traits are studied. Studying more traits from these populations will allow for the stacking of various QTLs for desirable traits.</p>
    </div>
</div></li>
<li><!-- Entry bib key -->
<div id="Hochhaus2023">
  
  <!-- Title -->
  <div class="title">Meta-Analysis of Rose Rosette Disease-Resistant Quantitative Trait Loci and a Search for Candidate Genes</div>
  <!-- Author -->
  <div class="author">Hochhaus, Tessa,&nbsp;Lau, Jeekin,&nbsp;Taniguti, Cristiane H.,&nbsp;Young, Ellen L.,&nbsp;Byrne, David H.,&nbsp;and Riera-Lizarazu, Oscar
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Pathogens, vol. 12, Apr, 2023
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.3390/pathogens12040575" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Rose rosette disease (RRD), caused by the rose rosette emaravirus (RRV), is a major viral disease in roses (Rosa sp.) that threatens the rose industry. Recent studies have revealed quantitative trait loci (QTL) for reduced susceptibility to RRD in the linkage groups (LGs) 1, 5, 6, and 7 in tetraploid populations and the LGs 1, 3, 5, and 6 in diploid populations. In this study, we seek to better localize and understand the relationship between QTL identified in both diploid and tetraploid populations. We do so by remapping the populations found in these studies and performing a meta-analysis. This analysis reveals that the peaks and intervals for QTL using diploid and tetraploid populations co-localized on LG 1, suggesting that these are the same QTL. The same was seen on LG 3. Three meta-QTL were identified on LG 5, and two were discovered on LG 6. The meta-QTL on LG 1, MetaRRD1.1, had a confidence interval (CI) of 10.53 cM. On LG 3, MetaRRD3.1 had a CI of 5.94 cM. MetaRRD5.1 had a CI of 17.37 cM, MetaRRD5.2 had a CI of 4.33 cM, and MetaRRD5.3 had a CI of 21.95 cM. For LG 6, MetaRRD6.1 and MetaRRD6.2 had CIs of 9.81 and 8.81 cM, respectively. The analysis also led to the identification of potential disease resistance genes, with a primary interest in genes localized in meta-QTL intervals on LG 5 as this LG was found to explain the greatest proportion of phenotypic variance for RRD resistance. The results from this study may be used in the design of more robust marker-based selection tools to track and use a given QTL in a plant breeding context.</p>
    </div>
</div></li>
<li><!-- Entry bib key -->
<div id="Taniguti2023">
  
  <!-- Title -->
  <div class="title">Developing best practices for genotyping-by-sequencing analysis in the construction of linkage maps</div>
  <!-- Author -->
  <div class="author">Taniguti, Cristiane Hayumi,&nbsp;Taniguti, Lucas Mitsuo,&nbsp;Amadeu, Rodrigo Rampazo,&nbsp;Lau, Jeekin,&nbsp;Gesteira, Gabriel de Siqueira,&nbsp;Oliveira, Thiago de Paula,&nbsp;Ferreira, Getulio Caixeta,&nbsp;Pereira, Guilherme da Silva,&nbsp;Byrne, David,&nbsp;Mollinari, Marcelo,&nbsp;Riera-Lizarazu, Oscar,&nbsp;and Garcia, Antonio Augusto Franco
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    GigaScience, vol. 12, pp. giad092, Oct, 2023
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="https://doi.org/10.1093/gigascience/giad092" class="btn btn-sm z-depth-0" role="button">URL</a>]
      [<a href="http://doi.org/10.1093/gigascience/giad092" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Genotyping-by-sequencing (GBS) provides affordable methods for genotyping hundreds of individuals using millions of markers. However, this challenges bioinformatic procedures that must overcome possible artifacts such as the bias generated by polymerase chain reaction duplicates and sequencing errors. Genotyping errors lead to data that deviate from what is expected from regular meiosis. This, in turn, leads to difficulties in grouping and ordering markers, resulting in inflated and incorrect linkage maps. Therefore, genotyping errors can be easily detected by linkage map quality evaluations.We developed and used the Reads2Map workflow to build linkage maps with simulated and empirical GBS data of diploid outcrossing populations. The workflows run GATK, Stacks, TASSEL, and Freebayes for single-nucleotide polymorphism calling and updog, polyRAD, and SuperMASSA for genotype calling, as well as OneMap and GUSMap to build linkage maps. Using simulated data, we observed which genotype call software fails in identifying common errors in GBS sequencing data and proposed specific filters to better handle them. We tested whether it is possible to overcome errors in a linkage map using genotype probabilities from each software or global error rates to estimate genetic distances with an updated version of OneMap. We also evaluated the impact of segregation distortion, contaminant samples, and haplotype-based multiallelic markers in the final linkage maps. Through our evaluations, we observed that some of the approaches produce different results depending on the dataset (dataset dependent) and others produce consistent advantageous results among them (dataset independent).We set as default in the Reads2Map workflows the approaches that showed to be dataset independent for GBS datasets according to our results. This reduces the number of required tests to identify optimal pipelines and parameters for other empirical datasets. Using Reads2Map, users can select the pipeline and parameters that best fit their data context. The Reads2MapApp shiny app provides a graphical representation of the results to facilitate their interpretation.</p>
    </div>
</div></li></ol>
<h2 class="bibliography">2022</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Taniguti2022">
  
  <!-- Title -->
  <div class="title">VIEWpoly - a visualization tool to integrate and explore results of polyploid genetic analysis</div>
  <!-- Author -->
  <div class="author">Taniguti, Cristiane Hayumi,&nbsp;Siqueira Gesteira, Gabriel,&nbsp;Lau, Jeekin,&nbsp;Silva Pereira, Guilherme,&nbsp;Zeng, Zhao-Bang,&nbsp;Byrne, David,&nbsp;Riera-Lizarazu, Oscar,&nbsp;and Mollinari, Marcelo
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Journal of Open Source Software, vol. 7, pp. 4242, Jun, 2022
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.21105/joss.04242" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Autopolyploid species comprise a significant part of the agronomic market. Some examples are potato, sweet potato, roses, and blueberries. They all present more than two copies of their entire genome, which increase the complexity of genetic models. Advancements in computational tools for linkage and quantitative trait loci (QTL) analysis in autopolyploids have allowed the in-depth exploration and understanding of their genetics. Despite providing advanced methods, the results of these tools may be challenging to interpret and connect to other sources of genomic information and consequently apply them in practical breeding applications. VIEWpoly is an R package for visualizing, exploring, and integrating results from different polyploid computational tools using an interactive graphical user interface. The package allows users to explore linkage and QTL analysis results and integrates these with genomic information in a genome browser, facilitating the search for candidate genes. In addition, it provides features to download comprehensive tables and graphics for further analysis and presentation. VIEWpoly is freely available as an R package at https://CRAN.R-project.org/package=viewpoly.</p>
    </div>
</div></li>
<li><!-- Entry bib key -->
<div id="Raverdy2022">
  
  <!-- Title -->
  <div class="title">Linkage Mapping of Biomass Production and Composition Traits in a Miscanthus sinensis Population</div>
  <!-- Author -->
  <div class="author">Raverdy, Raphael,&nbsp;Lourgant, Kristelle,&nbsp;Mignot, Emilie,&nbsp;Arnoult, Stephanie,&nbsp;Bodineau, Guillaume,&nbsp;Griveau, Yves,&nbsp;Taniguti, Cristiane H.,&nbsp;and Brancourt-Hulmel, Maryse
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Bioenergy Research, vol. 15, pp. 755-776, Jun, 2022
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.1007/s12155-022-10402-8" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Breeding miscanthus for biomass production and composition is essential for targeting high-yielding genotypes suited to different end-uses. Our objective was to understand the genetic basis of these traits in M. sinensis, according to different plant ages and environmental conditions. A diploid population was established in two locations according to a staggered-start design, which distinguished the plant age effect from climatic condition effect. An integrated genetic map of 2602 SNP markers distributed across 19 LGs was aligned with the M. sinensis reference genome and spanned 2770 cM. The QTL mapping was based on best linear unbiased predictions estimated across three climatic conditions and at least three ages in both locations. A total of 260 and 283 QTL were related to biomass production and composition traits, respectively. In each location, 40–60% were related to biomass production traits and stable across different climatic conditions and ages and 30% to biomass composition traits. Twelve QTL clusters were established based on either biomass production or composition traits and validated by high genetic correlations between the traits. Sixty-two putative M. sinensis genes, related to the cell wall, were evidenced in the QTL clusters of biomass composition traits and orthologous to those of sorghum and maize. Twelve of them were differentially expressed and belonged to gene families related to the cell wall biosynthesis identified in other miscanthus studies. These stable QTL constitute new insights into marker-assisted selection (MAS) breeding while offering a joint improvement of biomass production and composition traits.</p>
    </div>
</div></li></ol>
<h2 class="bibliography">2021</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Almeida2021">
  
  <!-- Title -->
  <div class="title">Angular Leaf Spot Resistance Loci Associated With Different Plant Growth Stages in Common Bean</div>
  <!-- Author -->
  <div class="author">Almeida, Caléo Panhoca,&nbsp;Carvalho Paulino, Jean Fausto,&nbsp;Bonfante, Gabriel Francesco Janini,&nbsp;Perseguini, Juliana Morini Kupper Cardoso,&nbsp;Santos, Isabella Laporte,&nbsp;Gonçalves, João Guilherme Ribeiro,&nbsp;Patrício, Flávia Rodrigues Alves,&nbsp;Taniguti, Cristiane Hayumi,&nbsp;Siqueira Gesteira, Gabriel,&nbsp;Garcia, Antônio Augusto Franco,&nbsp;Song, Qijian,&nbsp;Carbonell, Sérgio Augusto Morais,&nbsp;Chiorato, Alisson Fernando,&nbsp;and Benchimol-Reis, Luciana Lasry
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Frontiers in Plant Science, vol. 12, Apr, 2021
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.3389/fpls.2021.647043" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Angular leaf spot (ALS) is a disease that causes major yield losses in the common bean crop. Studies based on different isolates and populations have already been carried out to elucidate the genetic mechanisms of resistance to ALS. However, understanding of the interaction of this resistance with the reproductive stages of common bean is lacking. The aim of the present study was to identify ALS resistance loci at different plant growth stages (PGS) by association and linkage mapping approaches. An BC2F3 inter-gene pool cross population (AND 277 × IAC-Milênio – AM population) profiled with 1,091 SNPs from genotyping by sequencing (GBS) was used for linkage mapping, and a carioca diversity panel (CDP) genotyped by 5,398 SNPs from BeadChip assay technology was used for association mapping. Both populations were evaluated for ALS resistance at the V2 and V3 PGSs (controlled conditions) and R8 PGS (field conditions). Different QTL (quantitative trait loci) were detected for the three PGSs and both populations, showing a different quantitative profile of the disease at different plant growth stages. For the three PGS, multiple interval mapping (MIM) identified seven significant QTL, and the Genome-wide association study (GWAS) identified fourteen associate SNPs. Several loci validated regions of previous studies, and Phg-1, Phg-2, Phg-4, and Phg-5, among the 5 loci of greatest effects reported in the literature, were detected in the CDP. The AND 277 cultivar contained both the Phg-1 and the Phg-5 QTL, which is reported for the first time in the descendant cultivar CAL143 as ALS10.1UC. The novel QTL named ALS11.1AM was located at the beginning of chromosome Pv11. Gene annotation revealed several putative resistance genes involved in the ALS response at the three PGSs, and with the markers and loci identified, new specific molecular markers can be developed, representing a powerful tool for common bean crop improvement and for gain in ALS resistance.</p>
    </div>
</div></li></ol>
<h2 class="bibliography">2020</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Gazaffi2020a">
  
  <!-- Title -->
  <div class="title">fullsibQTL: An R package for QTL mapping in biparental populations of outcrossing species</div>
  <!-- Author -->
  <div class="author">Gazaffi, R.,&nbsp;Amadeu, R.R.,&nbsp;Mollinari, M.,&nbsp;Rosa, J.R.B.F.,&nbsp;Taniguti, C.H.,&nbsp;Margarido, G.R.A.,&nbsp;and Garcia, A.A.F.
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    2020
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://doi.org/10.1101/2020.12.04.412262" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>The copyright holder for this preprint (which was not certified by peer review) is the author/funder, who has granted bioRxiv a license to display the preprint in perpetuity. It is made available under aCC-BY-NC-ND 4.0 International license. Accurate QTL mapping in outcrossing species requires software programs which consider genetic features of these populations, such as markers with different segregation patterns and different level of information. Although the available mapping procedures to date allow inferring QTL position and effects, they are mostly not based on multilocus genetic maps. Having a QTL analysis based in such maps is crucial since they allow informative markers to propagate their information to less informative intervals of the map. We developed fullsibQTL, a novel and freely available R package to perform composite interval QTL mapping considering outcrossing populations and markers with different segregation patterns. It allows to estimate QTL position, effects, segregation patterns, and linkage phase with flanking markers. Additionally, several statistical and graphical tools are implemented, for straightforward analysis and interpretations. fullsibQTL is an R open source package with C and R source code (GPLv3). It is multiplatform and can be installed from https://github.com/augusto-garcia/fullsibQTL.</p>
    </div>
</div></li></ol>
<h2 class="bibliography">2018</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Conson2018">
  
  <!-- Title -->
  <div class="title">High-Resolution Genetic Map and QTL Analysis of Growth-Related Traits of Hevea brasiliensis Cultivated Under Suboptimal Temperature and Humidity Conditions</div>
  <!-- Author -->
  <div class="author">Conson, André R. O.,&nbsp;Taniguti, Cristiane H.,&nbsp;Amadeu, Rodrigo R.,&nbsp;Andreotti, Isabela A. A.,&nbsp;Souza, Livia M.,&nbsp;Santos, Luciano H. B.,&nbsp;Rosa, João R. B. F.,&nbsp;Mantello, Camila C.,&nbsp;Silva, Carla C.,&nbsp;Junior, Erivaldo José Scaloppi,&nbsp;Ribeiro, Rafael V,&nbsp;Guen, Vincent Le,&nbsp;Garcia, Antonio A. F.,&nbsp;Souza Gonçalves, Paulo,&nbsp;and Souza, Anete P.
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Frontiers in Plant Science, vol. 9, pp. 1-16, Aug, 2018
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="https://doi.org/10.3389/fpls.2018.01255 https://www.frontiersin.org/article/10.3389/fpls.2018.01255/full" class="btn btn-sm z-depth-0" role="button">URL</a>]
      [<a href="http://doi.org/10.3389/fpls.2018.01255" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>Rubber tree (Hevea brasiliensis) cultivation is the main source of natural rubber worldwide and has been extended to areas with suboptimal climates and lengthy drought periods; this transition affects growth and latex production. High-density genetic maps with reliable markers support precise mapping of quantitative trait loci (QTL), which can help reveal the complex genome of the species, provide tools to enhance molecular breeding, and shorten the breeding cycle. In this study, QTL mapping of the stem diameter, tree height, and number of whorls was performed for a full-sibling population derived from a GT1 and RRIM701 cross. A total of 225 simple sequence repeats (SSRs) and 186 single-nucleotide polymorphism (SNP) markers were used to construct a base map with 18 linkage groups and to anchor 671 SNPs from genotyping by sequencing (GBS) to produce a very dense linkage map with small intervals between loci. The final map was composed of 1,079 markers, spanned 3,779.7 cM with an average marker density of 3.5 cM, and showed collinearity between markers from previous studies. Significant variation in phenotypic characteristics was found over a 59-month evaluation period with a total of 38 QTLs being identified through a composite interval mapping method. Linkage group 4 showed the greatest number of QTLs (7), with phenotypic explained values varying from 7.67 to 14.07%. Additionally, we estimated segregation patterns, dominance, and additive effects for each QTL. A total of 53 significant effects for stem diameter were observed, and these effects were mostly related to additivity in the GT1 clone. Associating accurate genome assemblies and genetic maps represents a promising strategy for identifying the genetic basis of phenotypic traits in rubber trees. Then, further research can benefit from the QTLs identified herein, providing a better understanding of the key determinant genes associated with growth of Hevea brasiliensis under limiting water conditions.</p>
    </div>
</div></li></ol>
<h2 class="bibliography">2016</h2>
<ol class="bibliography"><li><!-- Entry bib key -->
<div id="Nakamura2016a">
  
  <!-- Title -->
  <div class="title">Reference genes for accessing differential expression among developmental stages and analysis of differential expression of OBP genes in Anastrepha obliqua</div>
  <!-- Author -->
  <div class="author">Nakamura, A.M. Aline Minali,&nbsp;Chahad-Ehlers, Samira,&nbsp;Lima, Andre Luis A.L.A.,&nbsp;Taniguti, C.H. Cristiane Hayumi C.H. Cristiane Hayumi,&nbsp;Sobrinho, I.,&nbsp;Torres, Felipe Rafael F.R. Felipe Rafael,&nbsp;Brito, Reinaldo Alves R.A. De,&nbsp;Jr., Iderval Sobrinho,&nbsp;Torres, Felipe Rafael F.R. Felipe Rafael,&nbsp;Brito, Reinaldo Alves R.A. De,&nbsp;Sobrinho, I.,&nbsp;Torres, Felipe Rafael F.R. Felipe Rafael,&nbsp;Brito, Reinaldo Alves R.A. De,&nbsp;Jr., Iderval Sobrinho,&nbsp;Torres, Felipe Rafael F.R. Felipe Rafael,&nbsp;and Brito, Reinaldo Alves R.A. De
  </div>

  <!-- Journal/Book title and date -->
  
  <div class="third-line">
    Scientific Reports, vol. 6, pp. 17480, 2016
  </div>

  <!-- Links/Buttons -->
  <div class="links">
      [<a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>]
      [<a href="http://www.nature.com/articles/srep17480" class="btn btn-sm z-depth-0" role="button">URL</a>]
      [<a href="http://doi.org/10.1038/srep17480" class="btn btn-sm z-depth-0" role="button">DOI</a>]
  </div>

  <!-- Hidden abstract block -->
    <div class="abstract hidden">
      <p>2016, Nature Publishing Group. All rights reserved.The West Indian fruit fly, Anastrepha obliqua, is an important agricultural pest in the New World. The use of pesticide-free methods to control invasive species such as this reinforces the search for genes potentially useful in their genetic control. Therefore, the study of chemosensory proteins involved with a range of responses to the chemical environment will help not only on the understanding of the species biology but may also help the development of environmentally friendly pest control strategies. Here we analyzed the expression patterns of three OBP genes, Obp19d-2, Obp56a and Obp99c, across different phases of A. obliqua development by qPCR. In order to do so, we tested eight and identified three reference genes for data normalization, rpl17, rpl18 and ef1a, which displayed stability for the conditions here tested. All OBPs showed differential expression on adults and some differential expression among adult stages. Obp99c had an almost exclusive expression in males and Obp56a showed high expression in virgin females. Thereby, our results provide relevant data not only for other gene expression studies in this species, as well as for the search of candidate genes that may help in the development of new pest control strategies.</p>
    </div>
</div></li></ol>
</div>

<!-- Footer -->
<footer>
  <div class="footer">
    Copyright © 2022
    <a href=""></a>.
    Powered by Jekyll with
    <a href="https://github.com/chrjabs/Grape-Academic-Theme">Grape Academic Theme</a>.
  </div>
</footer>

  </div>
</body>

</html>